Makefile

```diff
@@ -451,6 +453,47 @@ st-checks:
 	# running on the host.
 	iptables-save | grep -q 'calico-st-allow-etcd' || iptables $(IPT_ALLOW_ETCD)
 
+## Get the kubeadm-dind-cluster script
+tests/k8st/dind-cluster.sh:
+	wget -O tests/k8st/dind-cluster.sh https://raw.githubusercontent.com/neiljerram/kubeadm-dind-cluster/master/dind-cluster.sh
+	chmod +x tests/k8st/dind-cluster.sh
+
+.PHONY: k8s-test
+## Run the k8s tests
+k8s-test: $(NODE_CONTAINER_CREATED) calico_test.created tests/k8st/dind-cluster.sh k8s-stop k8s-start
+	docker run \
+	    -v $(CURDIR):/code \
+	    -v /var/run/docker.sock:/var/run/docker.sock \
+	    -v /home/$(USER)/.kube/config:/root/.kube/config \
+	    -v /home/$(USER)/.kubeadm-dind-cluster:/root/.kubeadm-dind-cluster \
+	    --privileged \
+	    --net host \
+        $(TEST_CONTAINER_NAME) \
+	    sh -c 'cp /root/.kubeadm-dind-cluster/kubectl /bin/kubectl && ls -ltr /bin/kubectl && which kubectl && cd /code/tests/k8st && nosetests -v --with-xunit --xunit-file="/code/report/k8s-tests.xml" --with-timer'
+	$(MAKE) k8s-stop
+
+.PHONY: k8s-start
+## Start k8s cluster
+k8s-start: tests/k8st/dind-cluster.sh calico-node.tar
+	CNI_PLUGIN=custom POD_CIDR=192.168.0.0/16 SKIP_SNAPSHOT=y tests/k8st/dind-cluster.sh up
+	kubectl apply -f  https://docs.projectcalico.org/master/getting-started/kubernetes/installation/hosted/etcd.yaml
+	docker cp calico-node.tar kube-master:/calico-node.tar
+	docker cp calico-node.tar kube-node-1:/calico-node.tar
+	docker cp calico-node.tar kube-node-2:/calico-node.tar
+	docker exec kube-master docker load -i /calico-node.tar
+	docker exec kube-node-1 docker load -i /calico-node.tar
+	docker exec kube-node-2 docker load -i /calico-node.tar
+	wget https://docs.projectcalico.org/master/getting-started/kubernetes/installation/hosted/calico.yaml -O - | \
+		sed s,quay.io/calico/node:master,$(BUILD_IMAGE):latest-$(ARCH), | \
```

Comment:
I wouldn't rely on the `quay` repo here - this is likely to change in a future release. If we could do a sed that handles varying repository names that would be better.

Reply:
This change is in the kubeadm-dind-cluster repo now - but I will improve it there.